client_src/js/utils/validators/api.validator.js

```diff
@@ -0,0 +1,23 @@
+import { ClientError } from "../../exceptions/client-error.handler";
+import { ClientErrors } from "../../exceptions/client.exceptions";
+
+function validateServerResponse(response) {
+  // Don't really like this, there may be a better way but currently the server re-directs everything and was hard to detect otherwise.
+  let isString = Object.prototype.toString.call(response) === "[object String]";
+
+  return !(isString && response.includes("<!DOCTYPE html>"));
+}
+
+function validatePath(pathname) {
+  // if (!pathname) throw new ClientError(ClientErrors.NO_PATHNAME_SUPPLIED);
+  let url = pathname;
+  if (!pathname?.includes("http")) {
```

Comment:
I did it this way because it's literally the only requirement for the client. We don't call ws/wss/mongodb currently. 

The http check is because currently I have to deal with /api/blabla endpoints for OctoFarm and OctoPrint with http:// endpoints. This way allowed for both to respond with .href, as for changing and refactoring this I haven't been able to test any OctoPrint connections on Canary to refine this further due to the connection bug.